app/src/main/java/com/keylesspalace/tusky/components/report/adapter/StatusViewHolder.kt

```diff
@@ -0,0 +1,166 @@
+import com.keylesspalace.tusky.util.StatusViewHelper.Companion.NO_INPUT_FILTER
+import kotlinx.android.synthetic.main.item_report_status.view.*
+import java.util.*
+
+class StatusViewHolder(itemView: View,
+                       private val useAbsoluteTime: Boolean,
+                       private val mediaPreviewEnabled: Boolean,
+                       private val viewState: StatusViewState,
+                       private val adapterHandler: AdapterHandler,
+                       private val getStatusForPosition: (Int) -> Status?) : RecyclerView.ViewHolder(itemView) {
+    private val mediaViewHeight = itemView.context.resources.getDimensionPixelSize(R.dimen.status_media_preview_height)
+    private val statusViewHelper = StatusViewHelper(itemView)
+
+    private val previewListener = object : StatusViewHelper.MediaPreviewListener {
+        override fun onViewMedia(v: View?, idx: Int) {
+            status()?.let { status ->
+                adapterHandler.showMedia(v, status, idx)
+            }
+        }
+
+        override fun onContentHiddenChange(isShowing: Boolean) {
+            status()?.id?.let { id ->
+                viewState.setMediaShow(id, isShowing)
+            }
+        }
+    }
+
+    init {
+        itemView.statusSelection.setOnCheckedChangeListener { _, isChecked ->
+            status()?.let { status ->
+                adapterHandler.setStatusChecked(status, isChecked)
+            }
+        }
+    }
+
+    fun bind(status: Status) {
+        itemView.statusSelection.isChecked = adapterHandler.isStatusChecked(status.id)
+
+        updateTextView()
+
+        val sensitive = status.sensitive
+
+        statusViewHelper.setMediasPreview(mediaPreviewEnabled, status.attachments, sensitive, previewListener,
+                viewState.isMediaShow(status.id, status.sensitive),
+                mediaViewHeight)
+
+        statusViewHelper.setupPollReadonly(status.poll, status.emojis, useAbsoluteTime)
+        setCreatedAt(status.createdAt)
+    }
+
+    private fun updateTextView() {
+        status()?.let { status ->
+            setupCollapsedState(status.isCollapsible(), viewState.isCollapsed(status.id, true),
+                    viewState.isContentShow(status.id, status.sensitive), status.spoilerText)
+
+            if (status.spoilerText.isBlank()) {
+                setTextVisible(true, status.content, status.mentions, status.emojis, adapterHandler)
+                itemView.statusContentWarningButton.visibility = View.GONE
+                itemView.statusContentWarningDescription.visibility = View.GONE
+            } else {
+                val emojiSpoiler = CustomEmojiHelper.emojifyString(status.spoilerText, status.emojis, itemView.statusContentWarningDescription)
+                itemView.statusContentWarningDescription.text = emojiSpoiler
+                itemView.statusContentWarningDescription.visibility = View.VISIBLE
+                itemView.statusContentWarningButton.visibility = View.VISIBLE
```

Comment:
there are extension functions `show()` and `hide()` to make this more Kotlin-like

Reply:
Done